kratos/sources/kratos_application.cpp

```diff
@@ -318,6 +318,20 @@ void KratosApplication::RegisterKratosCore() {
     RegisterVoxelMesherOperation();
 }
 
+
+void KratosApplication::DeregisterVariables() {
+    const std::string path = "variables."+mApplicationName;
+    if (Registry::HasItem(path)) {
+        auto& r_variables = Registry::GetItem(path);
+        // Iterate over items at path. For each item, remove it from the mappers.all branch too
+        for (auto i_key = r_variables.KeyConstBegin(); i_key != r_variables.KeyConstEnd(); ++i_key) {
+            Registry::RemoveItem("variables.all."+*i_key);
+        }
+        Registry::RemoveItem(path);
+    }
+}
```

Comment:
This removes the variables registered by the application, hence we don't allow registering same variable from multiple apps. So, I would as mentioned in the earlier comment, throw an error if the same variable name is registered from multiple places.

Reply:
Please note that the scope of this PR is very narrow (it is intended only for testing, it is not a general "unload application" mechanism). What I am doing here is just ensuring that whoever registered the variable takes care of de-registering it, not ensuring that the kernel remains in a complete state after application de-registry. I understand that this is consistent with what the de-registration logic does for other types of objects.